jb/itch5/compute_base.hpp

```diff
@@ -0,0 +1,247 @@
+#ifndef jb_itch5_compute_base_hpp
+#define jb_itch5_compute_base_hpp
+
+#include <jb/itch5/order_book_depth.hpp>
+#include <jb/itch5/add_order_message.hpp>
+#include <jb/itch5/add_order_mpid_message.hpp>
+#include <jb/itch5/order_cancel_message.hpp>
+#include <jb/itch5/order_delete_message.hpp>
+#include <jb/itch5/order_executed_message.hpp>
+#include <jb/itch5/order_executed_price_message.hpp>
+#include <jb/itch5/order_replace_message.hpp>
+#include <jb/itch5/stock_directory_message.hpp>
+#include <jb/itch5/unknown_message.hpp>
+
+#include <boost/functional/hash.hpp>
+#include <chrono>
+#include <functional>
+#include <unordered_map>
+
+namespace jb {
+namespace itch5 {
+
+/** 
+ * An implementation of jb::message_handler_concept to compute feed values.
+ *
+ * Keep a collection of all the order books, and forward the right
+ * updates to them as it handles the different message types in
+ * ITCH-5.0.
+ *
+ * Uses:
+ * The order_book_depth object to get the book depth
+ * Virtual calls check_callback() to allow the implementation 
+ * of derivate callback signatures and call conditions
+ */
+
+class compute_base {
+ public:
+  //@{
+  /**
+   * @name Type traits
+   */
+  /// Define the clock used to measure processing delays.
+  typedef std::chrono::steady_clock::time_point time_point;
+  /// Callbacks
+  typedef std::function<void(time_point,
+			     message_header const&,
+			     stock_t const&)
+			> callback_type;
+  //@}
+
+  /**
+   * The collection of order book depths.
+   */
+  typedef std::unordered_map<stock_t,              /**<symbol, key      */
+			     order_book_depth,     /**<The order book for that symbol */
+			     boost::hash<stock_t>  /**<hash function    */
+			    > books_by_security;
+    
+  /// Initialize an empty handler
+  compute_base() = default;
+
+  /// Initialize an empty handler and its call condition
+  compute_base(callback_type const& _callback)
+    : callback_(_callback),
+      orders_(),
+      books_() {};
+
+  virtual ~compute_base() {};
```

Comment:
That would be a pure virtual? correct? 
I do want to have object of the base class (maybe the name _base is misleading), like on ut_compute_inside.cpp to test it (for instance).
Maybe it should be?
virtual ~foo() = default;

(please note that this discussion is for the sake of my training, because I am going to rollback this refactoring)